File: extra_1/app/src/main/java/com/example/extra_1/model/WeatherModels.java
package com.example.extra_1.model;

import java.util.List;

public class WeatherModels {
    public static class ForecastResponse { public List<ForecastItem> list; }
    public static class ForecastItem { public String dt_txt; public MainInfo main; public List<WeatherInfo> weather; }
    public static class MainInfo { public double temp; }
    public static class WeatherInfo { public String description; }

    public static class DayForecast {
        public final String day;
        public final String tempText;
        public DayForecast(String day, String tempText) { this.day = day; this.tempText = tempText; }
    }
}

File: extra1/app/src/main/java/com/example/extra1/model/WeatherModels.kt
package com.example.extra1.model

data class ForecastResponse(val list: List<ForecastItem>)
data class ForecastItem(val dt_txt: String, val main: MainInfo, val weather: List<WeatherInfo>)
data class MainInfo(val temp: Double)
data class WeatherInfo(val description: String)

data class DayForecast(val day: String, val tempText: String)
